src/common/Assert.h

```diff
@@ -19,6 +20,4 @@
 // for now the asserts are always on!
 // TODO make the asserts primarity for DEBUG and google test.
 #define RelAssertMsg(exp, msg)                                                 \
-  (__builtin_expect(!(exp), 0)                                                 \
-       ? HandleAssertFail(#exp, __FILE__, __LINE__, msg)                       \
-       : (void)0)
+  (UNLIKELY(!(exp)) ? HandleAssertFail(#exp, __FILE__, __LINE__, msg) : (void)0)
```

Comment:
I don't like too much debug stuff in release builds, but if you do this

#ifndef NDEBUG
assert does something
#else
assert does nothing
#endif 
then the todo can be fixed immediately as NDEBUG is defined when building for Release